Button Block: Add draft page creation capability

dhruvikpatel18 opened the pull request:
Closes #56163

## What?
This PR adds the ability to create draft pages directly from the button block in the navigation editor, similar to how it already works in the navigation link block.

## Why?
Currently, users can create a draft page from the custom link block in the navigation editor if the page doesn't exist, but this functionality isn't available in the button block. This PR addresses this inconsistency by implementing the same feature in the button block, providing a more consistent editing experience.

## How?
1. Adds draft page creation functionality to the button block's LinkControl component
2. Imports necessary components and utilities from WordPress core
3. Adds permission checks to control when the draft creation feature is available
4. Uses the same pattern already implemented in the navigation link block

## Testing Instructions
1. Create a new post or page
2. Insert a navigation block
3. Add a button block to the navigation
4. Click on the link icon in the toolbar to edit the link
5. In the link search field, enter a page name that doesn't exist yet
6. Verify that an option appears to "Create draft page: [your search term]"
7. Click on this option and verify that a draft page is created and linked

## Screenshots or screencast <!-- if applicable -->

### Before
https://github.com/user-attachments/assets/9acbf1d2-4f18-41cf-b203-96a868228bb8

### After
https://github.com/user-attachments/assets/6c0b43ec-e4b8-4f8b-940a-6662ce8f6a0f

dhruvikpatel18 committed:
Add draft page creation capability to button block

Mamaduka reviewed (changes requested):
Thanks for contributing, @dhruvikpatel18!

I think we can reduce boilerplate code here by using the same approach as the Link format. See the [`InlineLinkUI` component](https://github.com/WordPress/gutenberg/blob/e093fefd041eb6cc4a4e7f67b92ab54fd75c8858/packages/format-library/src/link/inline.js#L264-L283) for details.

dhruvikpatel18 committed:
Refactor: Leveraged existing  settings to reduced boilerplate

dhruvikpatel18 commented:
> I think we can reduce boilerplate code here by using the same approach as the Link format. See the [`InlineLinkUI` component](https://github.com/WordPress/gutenberg/blob/e093fefd041eb6cc4a4e7f67b92ab54fd75c8858/packages/format-library/src/link/inline.js#L264-L283) for details.

@Mamaduka Thanks for the feedbacks, I have refactored the code and reduced boilerplate code by leveraging existing `LinkControl`.

Mamaduka commented on packages/block-library/src/button/edit.js:
```diff
@@ -219,6 +263,64 @@ function ButtonEdit( props ) {
+			}
+
+			const { getSettings } = select( blockEditorStore );
+			const _settings = getSettings();
```
```suggestion
			const _settings = select( blockEditorStore ).getSettings();
```

Suggestion (non-blocking): This could be a one liner.

Mamaduka commented on packages/block-library/src/button/edit.js:
```diff
@@ -400,6 +479,16 @@ function ButtonEdit( props ) {
+							}
+							withCreateSuggestion={ userCanCreatePages }
+							createSuggestionButtonText={ createButtonText }
+							showInitialSuggestions
+							suggestionsQuery={ getSuggestionsQuery(
+								'page',
+								'post-type'
+							) }
```
```suggestion
```

Let's disable initial suggestions for now and only display them when the user starts typing/searching.

Mamaduka reviewed (commented):
Thanks for the follow-up, @dhruvikpatel18!

Do you mind rebasing this branch on top of the latest trunk? I've also left two suggestions.

Mamaduka commented:
Hi, @dhruvikpatel18

Do you have time to follow up on the remaining suggestions? Also, let's rebase this branch on top of the latest trunk.

dhruvikpatel18 committed:
Disabled initial suggestions

dhruvikpatel18 commented:
Hi @Mamaduka, I've implemented your suggestion and removed the showInitialSuggestions prop from the LinkControl component. I've also rebased my branch on top of the latest trunk. The changes are now ready for review. Thank you for your feedback!

Mamaduka reviewed (approved):
Thank you, @dhruvikpatel18!

The enhancement implementation looks good ✅
